packages/block-editor/src/layouts/flex.js

```diff
@@ -94,9 +94,6 @@ export default {
 		onChange,
 		layoutBlockSupport,
 	} ) {
-		if ( layoutBlockSupport?.allowSwitching ) {
-			return null;
-		}
```

Comment:
I think this early return condition needs to be fixed. It should probably return `null` when both `allowJustification` and `allowVerticalAlignment` are `false`.

@ntsekouras, can you confirm? I think you worked on the initial implementation of this feature.

Reply:
It's been so long and even by looking at old PRs, I'm still not 100% sure why this check was added. If I remember correctly at that time we explicitly didn't want to show the justification controls in inspector controls too. Having said that, back then there were fewer layout props and there was no vertical alignment.

The `allowJustification` prop was added specifically for [constraint layout](https://github.com/WordPress/gutenberg/pull/45326/files#r1092919276) but in docs says that can also be applied to `flex` and defaults to `true`. There are no checks for it now in `flex` layout, so I guess we need to take into account both in `inspectorControls` and `toolBarControls`.

So, I'm not sure whether the `allowSwitching` check is still needed, but the other checks (`allowVerticalAlignment, allowJustification`) should be added.

I'll cc @tellthemachines if she has more context.

Reply:
I don't have any further context but agree this check isn't the right one here. 

One thing I'm noticing now is that setting `allowJustification` to false doesn't actually disable the justification controls for flex layout either in the toolbar or the sidebar. That should also be addressed, but not necessarily as part of this PR. 

Given that this change by itself fixes #67016 and doesn't affect anything else, I'm ok with merging this as is and fixing the justification controls separately.